horace_core/sqw/@sqw/unary_op_manager.m

```diff
@@ -0,0 +1,23 @@
+function w = unary_op_manager(w1, unary_op)
+% Implement unary arithmetic operations for objects containing a signal and variance arrays.
+
+% Generic method edited for sqw class. Must have
+%   (1) have methods to set, get and find size of signal and variance arrays:
+%           >> sz = sigvar_size(obj)
+%           >> w = sigvar(obj)          % w is sigvar object (has fields w.s, w.e)
+%           >> obj = sigvar_set(obj,w)  % w is sigvar object
+%   (2) have dimensions method that gives the dimensionality of the double array
+%           >> nd = dimensions(obj)
+%   (3) have private function that returns class name
+%           >> name = classname     % no argument - gets called by its association with the class
+
+w = w1;
+for i=1:numel(w1)
+    if ~isempty(w1(i).data.pix)
+        w(i).data.pix = w(i).data.pix.do_unary_op(unary_op);
```

Comment:
Is this going to be in the common superclass for sqw and dnd? So pix is not empty for sqw and is empty for dnd? Or are there scenarios where w1(i) is sqw but pix is empty? In first case maybe comment appropriately to facilitate the port to dnd. In second case a useful reminder of what an empty pix sqw is.

Reply:
There is a common base class, and it may be that the `unary_op_manager` can be moved up to it.

This `if` is handling the case where an SQW has had the PixelData removed (which as Harry pointed out can be done). My initial implementation had an `isa(w1(i), 'sqw')` check there which was always true and resulted in errors if the an empty pixeldata object was passed - there is a test specifically for this use-case.